### PR TITLE
Fix add book replaces existing book

### PR DESCRIPTION
**Pull request by amit1307**
## Summary of change

Fix an issue related to adding a book replaces an existing book.

## Related issue

https://github.com/knjk04/book-project/issues/240

### CODE REVIEW

**Comment by knjk04** [comment]
Thanks! I'll review this shortly

**Comment by knjk04** [comment]
As you may have seen on our Gitter group chat, we've moved the repository from my personal account to an organisation. Could you update the remote for your local fork, please?

$ git remote set-url origin https://github.com/Project-Books/book-project

File: src/test/java/com/karankumar/bookproject/ui/book/BookFormTest.java [hunk 1]
```diff
@@ -599,6 +599,26 @@ void shouldAddBookToDatabaseWhenSaveEventIsCalled() {
...
 
+    @Test
+    void shouldAddBooksToDatabaseWhenSaveEventIsCalled_withoutReplacingExistingBook() {
+        // given
+        bookForm = createBookForm(TO_READ, false);
+        bookForm.addListener(BookForm.SaveEvent.class, event -> bookService.save(event.getBook()));
+        bookForm.saveButton.click();
+
+        // when
+        bookTitle = "someOtherBook";
```
Comment (knjk04) [inline src/test/java/com/karankumar/bookproject/ui/book/BookFormTest.java:610]: I wouldn't recommend changing the name of the global variable and not making it `final` without resetting it back (or some other way). 

We use the `bookTitle` in assertions, so it's best to keep it to a value that's known (in case the tests run in a different order).

File: src/test/java/com/karankumar/bookproject/ui/book/BookFormTest.java [hunk 2]
```diff
@@ -672,6 +692,8 @@ void shouldUpdateValuesWhenBookIsMovedBetweenShelves(PredefinedShelf.ShelfName i
         correctBookAttributesPresent(newShelf, bookInDatabase);
     }
 
+
+
```
Comment (knjk04) [inline src/test/java/com/karankumar/bookproject/ui/book/BookFormTest.java:696]: Could you please undo this change?

**Review by knjk04 — CHANGES_REQUESTED** [review]
Thanks! There's just a couple of minor things to address

File: src/test/java/com/karankumar/bookproject/ui/book/BookFormTest.java [hunk 3]
```diff
@@ -599,6 +599,26 @@ void shouldAddBookToDatabaseWhenSaveEventIsCalled() {
...
+        // when
+        bookTitle = "someOtherBook";
+        bookForm = createBookForm(TO_READ, false);
+        bookForm.addListener(BookForm.SaveEvent.class, event -> bookService.save(event.getBook()));
+        bookForm.saveButton.click();
+
+        // then
+        Assertions.assertEquals(2, bookService.count());
+        List<Book> booksInDatabase = bookService.findAll();
+        Assertions.assertEquals("Superintelligence: Paths, Dangers, Strategies", booksInDatabase.get(0).getTitle());
```
[the remark this answers is not in the document]
Comment (amit1307) [reply src/test/java/com/karankumar/bookproject/ui/book/BookFormTest.java]: Done

File: src/test/java/com/karankumar/bookproject/ui/book/BookFormTest.java (hunk 1 above)
Comment (amit1307) [reply src/test/java/com/karankumar/bookproject/ui/book/BookFormTest.java]: Done

**Review by knjk04 — APPROVED** [review]
Thanks a lot! This bug was causing a lot of trouble!
